src/patchTypescript.ts

```diff
@@ -0,0 +1,39 @@
+// tslint:disable-next-line:no-implicit-dependencies
+import * as ts from 'typescript'; // Imported for types alone
+
+export interface TypeScriptPatchConfig {
+  /**
+   * Ususally, the compilerHost created with typescript.createWatchCompilerHost will bail out of diagnostics collection if there has been any syntactic error.
+   * (see https://github.com/Microsoft/TypeScript/blob/89386ddda7dafc63cb35560e05412487f47cc267/src/compiler/watch.ts#L141 )
+   * If this plugin is running with `checkSyntacticErrors: false`, this might lead to situations where no syntactic errors are reported within webpack
+   * (because the file causing a syntactic error might not get processed by ts-loader), but there are semantic errors that would be missed due to this behavior.
+   * This ensures that the compilerHost always assumes that there were no syntactic errors to be found and continues to check for semantic errors.
+   */
+  skipGetSyntacticDiagnostics: boolean;
+}
+
+export function patchTypescript(
+  typescript: typeof ts,
+  config: TypeScriptPatchConfig
+) {
+  const origTypescript = { ...typescript };
+
+  const patchSkipGetSyntacticDiagnostics: Pick<
+    typeof ts,
+    'createEmitAndSemanticDiagnosticsBuilderProgram'
+  > = {
+    createEmitAndSemanticDiagnosticsBuilderProgram(...args: any[]) {
+      const program = (origTypescript.createEmitAndSemanticDiagnosticsBuilderProgram as any)(
```

Comment:
I'm not sure if `this` inside `createEmitAndSemanticDiagnosticBuilderProgram` will be correct (as we have typescript and origTypescript). I would feel more comfortable with explicit `this` binding:
```
const program = originalCreateEmitAndSemanticDiagnosticsBuilderProgram.apply(typescript, args);
```

Reply:
From what I've seen, the typescript code doesn't really make use of the `this` keyword at all, they just have that one reference to `ts` and a lot of nested closures (which is why we can't just wrap it, but have to override some methods directly on the original object).

But yeah, apart from that, this is more future-proof if they should change that at any time in the future. I'll change it to `apply` :+1: